app/src/main/java/ingsw/group1/findmyphone/activity/MainActivity.java

```diff
@@ -117,6 +125,53 @@ public void requestPermissions() {
             ActivityCompat.requestPermissions(this, permissions, APP_PERMISSION_REQUEST_CODE);
     }
 
+    /***
+     * @author Turcato
+     * This method is executed both when the app is running or not.
+     * Based on the message's content, opens AlarmAndLocateResponseActivity if it's a request
+     * message,
+     * otherwise if it contains the location response (the only one expected) it opens the
+     * default maps application
+     * to the received location
+     *
+     * @param message Received SMSMessage class of SmsHandler library
+     */
+    public void onMessageReceived(SMSMessage message) {
+        manager.activeResponse(message, AlarmAndLocateResponseActivity.class);
```

Comment:
Both this and `onDestroy` should not be here, they interfere with the background receival

Reply:
This is a leftover from the old demos

Reply:
Yes, and they should be removed, because the service handles the receival, and must not be destroyed every time the activity is killed

Reply:
Ok, thanks. I have removed them.